api-tests/platform/drivers/watchdog/nrf/nrf_wdt.c

```diff
@@ -0,0 +1,134 @@
+__IM  uint32_t  RESERVED3[95];
+__IOM uint32_t  INTENSET;         /*!< (@ 0x00000304) Enable interrupt                             */
+__IOM uint32_t  INTENCLR;         /*!< (@ 0x00000308) Disable interrupt                            */
+__IM  uint32_t  RESERVED4[6];
+__IOM uint32_t  NMIENSET;         /*!< (@ 0x00000324) Enable interrupt                             */
+__IOM uint32_t  NMIENCLR;         /*!< (@ 0x00000328) Disable interrupt                            */
+__IM  uint32_t  RESERVED5[53];
+__IM  uint32_t  RUNSTATUS;        /*!< (@ 0x00000400) Run status                                   */
+__IM  uint32_t  REQSTATUS;        /*!< (@ 0x00000404) Request status                               */
+__IM  uint32_t  RESERVED6[63];
+__IOM uint32_t  CRV;              /*!< (@ 0x00000504) Counter reload value                         */
+__IOM uint32_t  RREN;             /*!< (@ 0x00000508) Enable register for reload request registers */
+__IOM uint32_t  CONFIG;           /*!< (@ 0x0000050C) Configuration register                       */
+__IM  uint32_t  RESERVED7[4];
+__OM  uint32_t  TSEN;             /*!< (@ 0x00000520) Task stop enable                             */
+__IM  uint32_t  RESERVED8[55];
+__OM  uint32_t  RR[8];            /*!< (@ 0x00000600) Description collection: Reload request n     */
+};          		   			  /*!< Size = 1568 (0x620)	                                       */
+
+/* Register: WDT_CONFIG */
+/* Description: Configuration register */
+#define WDT_CONFIG_STOPEN_Pos (6UL) /*!< Position of STOPEN field. */
+#define WDT_CONFIG_STOPEN_Disable (0UL) /*!< Do not allow stopping WDT */
+#define WDT_CONFIG_STOPEN_Enable (1UL) /*!< Allow stopping WDT */
+
+#define WDT_CONFIG_SLEEP_Pos (0UL) /*!< Position of SLEEP field. */
+#define WDT_CONFIG_SLEEP_Pause (0UL) /*!< Pause WDT while the CPU is sleeping */
+#define WDT_CONFIG_SLEEP_Run (1UL) /*!< Keep WDT running while the CPU is sleeping */
+
+/* Register: WDT_RUNSTATUS */
+/* Description: Run status */
+#define WDT_RUNSTATUS_RUNSTATUSWDT_Pos (0UL) /*!< Position of RUNSTATUSWDT field. */
+#define WDT_RUNSTATUS_RUNSTATUSWDT_Msk (0x1UL << WDT_RUNSTATUS_RUNSTATUSWDT_Pos) /*!< Bit mask of RUNSTATUSWDT field. */
+#define WDT_RUNSTATUS_RUNSTATUSWDT_NotRunning (0UL) /*!< Watchdog is not running */
+#define WDT_RUNSTATUS_RUNSTATUSWDT_Running (1UL) /*!< Watchdog is running */
+
+/**
+    @brief           - Initializes an hardware watchdog timer
+    @param           - base_addr       : Base address of the watchdog module
+                     - time_us         : Time in micro seconds
+    @return          - SUCCESS, FAILURE
+**/
+int nrf_wdt_init(addr_t base_addr, uint32_t time_us)
+{
+    struct NRF_WDT_Type *nrf_wdt = (struct NRF_WDT_Type*)base_addr;
+
+	/* From nRF53 documentation, the timeout in seconds is
+	 *       timeout [s] = (CRV + 1) / 32768
+	 */
+	nrf_wdt->CRV = (time_us / 1000000 * 32768) - 1;
```

Comment:
When time_us == 1 we would want CRV == 0.

But it is -1 here. The minimal CRV value is 0xF.

Also, I believe * has precedence over /, so we are dividing by 32k insead of multiplying by 32k. Even if I am wrong, let us add some parenthesis to not rely on obscure operator precedence.